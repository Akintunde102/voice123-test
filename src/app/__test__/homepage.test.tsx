import { render, screen } from "@testing-library/react";
import HomePage from '../page'

describe('Page', () => {
  it('renders a heading', () => {

    render(<HomePage />)

    const heading = screen.getByRole('heading', {
      level: 1,
      name: /voice123/i
    })

    expect(heading).toBeInTheDocument()
  })

  it('renders a submit button', () => {
    render(<HomePage />)

    const button = screen.getByRole('button', { name: /search/i })

    expect(button).toBeInTheDocument()
  })

  it('renders an input box', () => {
    render(<HomePage />)

    const input = screen.getByRole('textbox')

    expect(input).toBeInTheDocument()
  })


  it("renders homepage unchanged", () => {
    const { container } = render(<HomePage />);
    expect(container).toMatchSnapshot();
  });
})