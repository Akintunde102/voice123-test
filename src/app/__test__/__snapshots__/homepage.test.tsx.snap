// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders homepage unchanged 1`] = `
<div>
  <div
    class="MuiBox-root css-gds0zx"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1tgbrhe-MuiContainer-root"
    >
      <div
        class="MuiStack-root css-4i30ot-MuiStack-root"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-4713h1-MuiTypography-root"
        >
          Voice123
          <span
            class="MuiTypography-root MuiTypography-h1 css-13lepzk-MuiTypography-root"
          >
            Search
          </span>
        </h1>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-0.5 css-gxf7ul-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-9 css-173qa7i-MuiGrid-root"
          >
            <div
              aria-label="Search for Voice Actors"
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-hiddenLabel css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  aria-label="Search for Voice Actors"
                  autocomplete="off"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputHiddenLabel css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="outlined-basic"
                  placeholder="Cartoon Voice Actor"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ihdtdm"
                  >
                    <span
                      class="notranslate"
                    >
                      â€‹
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3 css-zbaamk-MuiGrid-root"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-sghohy-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="SearchIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14"
                  />
                </svg>
              </span>
              Search
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
